### PR TITLE
[ga] Add Irish

### PR DESCRIPTION
**Pull request by jimregan**
Replaces #915

### CODE REVIEW

**Comment by jimregan** [comment]
@danielnaber can I get a review, please?

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/tagset.txt [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
++Abr	abbreviation, e.g. lch. (leathanach) ‘page’
+
+Table 16.           Copula Tags
+Tag	Description
++Cop	verb
++Sg +Pl	singular and plural
++Pres	present / future
++PresSubj	present subjunctive
++Past	past / conditional
++Dep	dependant clause
```
Comment (dpelle) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/tagset.txt]: dependant → dependent

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/tagset.txt [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
++Pro	relative particle with pronoun, e.g. gach a tharla ‘all that which happened’
+
+
+Table 18.	Speech Tags
+Tag	Description
++Cmc	communicator, e.g.mar a déarfá 'as you might say', an dtuigeann tú 'do you understand', just, you know, anyway, cos, because, fair play
++Event	non-speech event: laugh, cough, sneeze etc.
++Filler	filled pause, e.g. em, ah, mm, hm, etc.
++Fragment	phonetic fragments, e.g. bh- m- amá- etc.
++Xxx	indecipherable speech; word(s) not recogized or are inaudible
```
Comment (dpelle) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/tagset.txt]: recogized → recognized

**Comment by jimregan** [comment]
Thanks @dpelle

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/PeopleRule.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  }  
+
+  @Override
+  public final String getId() {
+    return "GA_PEOPLE";
+  }
+
+ @Override
+  public String getDescription() {
+    return "";
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/PeopleRule.java]: + desc

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/PrestandardReplaceRule.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  @Override
+  public final String getId() {
+    return "GA_PRESTANDARD_REPLACE";
+  }
+
+  @Override
+  public String getDescription() {
+    return "";
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/PrestandardReplaceRule.java]: + desc

File: languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/SpacesRule.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  }  
+
+  @Override
+  public final String getId() {
+    return "GA_SPASANNA";
+  }
+
+ @Override
+  public String getDescription() {
+    return "";
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/SpacesRule.java]: + desc

File: languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Jim O'Regan <jaoregan@tcd.ie>
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java]: Is this a different license (not LGPL) on purpose? Not that it's a problem, but keeping the licenses coherent would be nice.

File: languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+package org.languagetool.tagging.ga;
+
+public class Retaggable {
+  String word;
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java]: Can these be made private?

File: languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  String restrictToPos;
+  String appendTag;
+  String prefix;
+  public Retaggable(String word, String restrictToPos, String appendTag) {
+    this.word = word;
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java]: Objects.requireNotNull(word) would be nice (if it make sense)

File: languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Utils.java [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+        }
+      } else {
+        return Character.toString(in.charAt(0)) + outh + in.substring(1);
+      }
+    } else {
+      return in;
+    }
+  }
+
+  public static String eclipse(String in) {
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Utils.java]: Short javadoc would be nice to explain what "eclipse" mean in GA.

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/added.txt [hunk 10]
```diff
@@ -0,0 +1,76 @@
+atáid	bí	Verb:VI:PresInd:3P:Pl:Rel:CM
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/added.txt]: A comment added with `#` would be good, could refer to the English `added.txt` maybe

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/conv-gramadoir.pl [hunk 11]
```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/perl
+
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/conv-gramadoir.pl]: A one-line comment what this does would be nice.

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/disambiguation.xml [hunk 12]
```diff
@@ -0,0 +1,1378 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    Disambiguation Rules for LanguageTool
+    Copyright (C) 2016 Trinity College, Dublin
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/disambiguation.xml]: Could you add the license?

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/earraidi-ga.bs [hunk 13]
```diff
@@ -0,0 +1,4697 @@
+Abbey An Mhainistir
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/earraidi-ga.bs]: One-line comment and license would be great.

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/earraidi.txt [hunk 14]
```diff
@@ -0,0 +1,1999 @@
+Absolute=Absalóid
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/earraidi.txt]: One-line comment and license would be great.

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/filter-archaic.txt [hunk 15]
```diff
@@ -0,0 +1,45 @@
+*	*	Num:Op
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/filter-archaic.txt]: One-line comment and license would be great.

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/filter-gramadoir.pl [hunk 16]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/perl
+
```
Comment (danielnaber) [inline languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/filter-gramadoir.pl]: One-line comment and license would be great (also applied to some other files).

File: languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Utils.java (hunk 9 above)
Comment (jimregan) [reply languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Utils.java]: tl;dr - 'to apply eclipsis': there's a rather lengthy Wikipedia article about Irish initial mutations: https://en.wikipedia.org/wiki/Irish_initial_mutations
(oh, and I've added the javadoc)

File: languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/earraidi-ga.bs (hunk 13 above)
Comment (jimregan) [reply languagetool-language-modules/ga/src/main/resources/org/languagetool/resource/ga/earraidi-ga.bs]: I've deleted that (which I should've done about a year ago!)

File: languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java (hunk 6 above)
Comment (jimregan) [reply languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Retaggable.java]: Yes, the licence here (and in Utils.java) is on purpose: these are pieces that are quite general purpose in terms of processing Irish, and don't depend on anything else, so I'd like that anyone who wants to process Irish can use them without having to ask me.

**Comment by jimregan** [comment]
@danielnaber Thanks for the review; I think I've covered everything

**Comment by danielnaber** [comment]
@jimregan Great, I'll have a look. I think we can probably merge this in 3-4 days. We have just added Arabic and I want to wait a bit until that works properly.

**Comment by jimregan** [comment]
Ok, makes sense. Thanks

**Comment by danielnaber** [comment]
@jimregan Could you add a `common_words.txt` with the most 10,000 common Irish words? We use this for automatic language detection. See `en/common_words.txt` for an example. It's better if these words include colloquial words and words typically used in email (hello, hi, regards, ...)

**Comment by jimregan** [comment]
@danielnaber I've add common_words.txt, as well as ga.txt in eval/

File: languagetool-language-modules/ga/pom.xml [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+  <dependencies>
+    <dependency>
+      <groupId>org.languagetool</groupId>
+      <artifactId>languagetool-core</artifactId>
+      <version>${languagetool.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.languagetool</groupId>
+      <artifactId>hunspell-native-libs</artifactId>
```
Comment (danielnaber) [inline languagetool-language-modules/ga/pom.xml]: Is this still needed? We got rid of this dependency for all other languages recently (in 4.8).

Comment (jimregan) [reply languagetool-language-modules/ga/pom.xml]: No, it's not. Removed.

**Comment by danielnaber** [comment]
This PR has finally been merged. Thanks again for the contribution. Unless there's a problem (related or unrelated to this PR), the code will be live tonight at around midnight CET on languagetool.org (it will then also be available in the browser add-on). To use it, set language to "automatic" as long as there's no value in the drop-down.

**Comment by jimregan** [comment]
Great, thanks!

**Pull request merged**